core/src/test/java/org/ldaptive/dns/DNSDomainFunctionTest.java

```diff
@@ -0,0 +1,74 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.ldaptive.dns;
+
+import org.ldaptive.dn.Dn;
+import org.ldaptive.dn.NameValue;
+import org.ldaptive.dn.RDn;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Unit test for {@link DNSDomainFunction}.
+ *
+ * @author  Middleware Services
+ */
+public class DNSDomainFunctionTest
+{
+
+
+  /**
+   * DN test data.
+   *
+   * @return  test data
+   */
+  @DataProvider(name = "DNs")
+  public Object[][] createDNs()
+  {
+    return
+      new Object[][]{
+        new Object[]{
+          Dn.builder().add("cn=John Doe").add("ou=accounting").add("dc=example").add("dc=net").build(),
+          "example.net",
+        },
+        new Object[]{
+          Dn.builder().add("DC=ldaptive").add("DC=org").build(),
+          "ldaptive.org",
+        },
+        new Object[]{
+          Dn.builder().add("DC=ldap").add("ou=example").add("DC=ldaptive").add("DC=org").build(),
+          "ldaptive.org",
+        },
+        new Object[]{
+          Dn.builder()
+            .add("CN=test")
+            .add(new RDn(new NameValue("DC", "ldap1"), new NameValue("DC", "ldap2")))
```

Comment:
I suppose this test case hits the `rdn.size() == 1` condition. Could this be a valid way to say (either "ldap1.ldaptive.org" or "ldap2.ldaptive.org", but the base DN is still "ldaptive.org"? I guess I'm getting at would you expect to see this in the wild or is it more a matter of input validation?

Reply:
Not according to the draft:

> An RDN is able to be converted if it (1)
>    consists of a single AttributeTypeAndValue; (2) the attribute type
>    is "DC"; and (3) the attribute value is non-null

So no one should expect that behavior from a multi-value RDN.

Reply:
Based on the excerpt you cited, should you not throw in that case?

Reply:
Since non "DC" RDNs are to be expected and ignored, I choose to treat any multi-value RDNs the same. Throwing wouldn't be wrong, but it may result in a more complex implementation. For instance, would you throw for any multi-value RDN or just one containing "DC"?